Sources/SourceKitLSP/Swift/CodeActions/UpdateDocumentation.swift

```diff
@@ -0,0 +1,138 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftParser
+import SwiftRefactor
+import SwiftSyntax
+
+/// Insert a documentation template associated with a function.
+///
+/// ## Before
+///
+/// ```swift
+/// static func refactor(syntax: DeclSyntax, in context: Void) -> DeclSyntax? {}
+/// ```
+///
+/// ## After
+///
+/// ```swift
+/// ///
+/// /// - Parameters:
+/// ///   - syntax:
+/// ///   - context:
+/// /// - Returns:
+/// static func refactor(syntax: DeclSyntax, in context: Void) -> DeclSyntax? {}
+/// ```
+@_spi(Testing)
+public struct UpdateDocumentation: SyntaxRefactoringProvider {
+  @_spi(Testing)
+  public static func refactor(syntax: DeclSyntax, in context: Void) -> DeclSyntax? {
+    let hasDocumentation = syntax.leadingTrivia.contains(where: { trivia in
+      switch trivia {
+      case .blockComment(_), .docBlockComment(_), .lineComment(_), .docLineComment(_):
+        return true
+      default:
+        return false
+      }
+    })
+
+    guard !hasDocumentation else {
+      return nil
+    }
+
+    var content: [TriviaPiece] = [
+      .docLineComment("/// A description"),
+      .newlines(1),
+    ]
+
+    if let parameters = syntax.parameters?.parameters {
+      if parameters.count == 1, let firstParam = parameters.first {
+        let paramToken = (firstParam.secondName ?? firstParam.firstName).trimmed
```

Comment:
Using `.text` gives the underlying text of the token directly without having to create a new token by trimming the current one.

```suggestion
        let paramToken = firstParam.secondName.text ?? firstParam.firstName.text
```

Reply:
Sure